<launch>

  <arg name="arg1" default="true"/>
  <node pkg="fixed_hk_camera" type="fixed_hk_camera_node" name="fixed_hk_camera_node" respawn="false" output="screen">
      <param name="handleset_xy" type="double" value="0.0"/>
      <param name="handleset_z" type="double" value="0.0"/>
      <param name="camera_id" type="int" value="1"/>
      <param name="sdkcom_dir" value="/usr/local/lib/thirdparty_lib"/>
      <param name="heartbeat_topic_string" value="/yd/heartbeat"/>
      <param name="device_ip" value="192.168.1.66"/>
      <param name="device_port" value="8000"/>
      <param name="device_username" value="admin"/>
      <param name="device_password" value="abcd1234"/>
      <param name="pan_range" type="str" value="0,360"/>
      <param name="tilt_range" type="str" value="-5,90"/>       
      <param name="heartbeat_topic_string" type="str" value="/yd/heartbeat"/>
      <param name="set_infrared_focus" type="bool" value="true"/>
      <param name="set_infrared_focus_waittime" type="int" value="20"/>
      <param name="ptz_topic" value="/fixed/yuntai/position"/>
      <param name="ptz_server_name" value="/fixed/internal/platform_cmd"/>
      <param name="pub_raw_temp" type="bool" value="true"/>
      <param name="raw_temp_topic" value="/fixed/internal/temperature"/>
      <param name="auto_zoom" type="bool" value="$(arg arg1)"/>
  </node>

  <node pkg="fixed_hk_camera" type="visible_node" name="visible_node" respawn="false" output="screen">
      <param name="camera_id" type="int" value="1"/>
      <param name="cali_offset_h" type="int" value="90"/>
      <param name="cali_offset_v" type="int" value="0"/>
      <param name="h_offset" type="double" value="0.03"/>
      <param name="v_offset" type="double" value="0"/>
      <param name="auto_zoom" type="bool" value="$(arg arg1)"/>
      <param name="camera_image_width" type="int" value="1920"/>
      <param name="camera_image_height" type="int" value="1080"/>
      <param name="camera_cmos_width" type="double" value="5.28"/>
      <param name="camera_cmos_height" type="double" value="2.97"/> 
      <param name="camera_pixel_size" type="double" value="3.75"/> 
      <param name="camera_file" value="$(find fixed_hk_camera)/launch/66color_map0_-10.yaml"/>
      <param name="camera_focus_mm" type="str" value="4.3,8.6,12.9,17.2,21.5,25.8,30.1,34.4,38.7,43,47.3,51.6,55.9,60.2,64.5,
      68.8,73.1,77.4,81.7,86,90.3,94.6,98.9,103.2,107.5,111.8,116.1,120.4,124.7,129"/>
      <param name="camera_focus_value" type="str" value="0,4281,7007,8783,10014,10933,11643,12225,12715,13138,13518,13854,14161,14442,14700,14939,15147,15337,15508,15655,15784,15894,15986,16072,16145,16206,16262,16311,16353,16384"/>
  </node>

  <node pkg="fixed_hk_camera" type="infrared_node" name="infrared_node" respawn="false" output="screen">
      <param name="camera_id" type="int" value="1"/>
      <param name="cali_offset_h" type="int" value="90"/>
      <param name="cali_offset_v" type="int" value="0"/>
      <param name="camera_file" value="$(find fixed_hk_camera)/launch/66color_map0_-10.yaml"/>
  </node>
  
</launch>


